xarray/core/merge.py

```diff
@@ -748,10 +707,20 @@ def merge_core(
 
     _assert_compat_valid(compat)
 
+    objects = list(objects)
+    if skip_align_args is None:
+        skip_align_args = []
+
+    skip_align_objs = [(pos, objects.pop(pos)) for pos in skip_align_args]
+
     coerced = coerce_pandas_values(objects)
     aligned = deep_align(
         coerced, join=join, copy=False, indexes=indexes, fill_value=fill_value
     )
+
+    for pos, obj in skip_align_objs:
+        aligned.insert(pos, obj)
```

Comment:
Could we equivalently just not call `merge_core` with these skipped arguments? What does it mean to include a skipped argument here?

Reply:
It is used when merging `coords` and `data_vars` during the creation of a new Dataset:

`coords` has already been coerced to a `Coordinates` object and we want to exclude it from alignment (coordinate variables should already be aligned together and it shouldn't be re-indexed). But we still need to merge it with the `data_vars` mapping and align data variables with coordinate indexes.